src/mame/drivers/viper.cpp

```diff
@@ -75,27 +82,44 @@
-        mocapglf            Sensor I/O error
-        mocapb,j            Crash after self checks
-        p911                "Distribution error"
-        p911e,j,uc,kc       Hangs at POST, with IRQ3 it crashes at first 3d frame
-        p9112               RTC self check bad
+        boxingm             Goes in-game. Controllers are not emulated. Various graphical glitches.
+        jpark3              Goes in-game. Controllers are not emulated. Various graphical glitches.
+        mocapb,j            Goes in-game. Controllers are not emulated. Various graphical glitches. Random crashes.
+        ppp2nd,a            Fully playable with graphical glitches. No network or DVD support. Crashes when returning to game mode from test menu.
+        p911(all)           Goes in-game. Controllers are not emulated. Various graphical glitches.
+        tsurugi,j           Goes in-game. Controllers are not emulated. Various graphical glitches.
+
+        gticlub2,ea         Attract mode works. Coins up. Hangs in various places. Will crash with "network error" after stage is selected.
+        thrild2,a           Attract mode with partial graphics. Coins up. Hangs in various places.
+
+        sscopefh            Graphics heavily glitched. Gun controller is not emulated. Sensor error and hopper error stop it from working.
+
+        mfightc,c           Requires touch panel emulation. Gets stuck at "Waiting for central monitor, checking serial...".
+        xtrial              Hangs at "Please set the time for the bookkeeping" message.
+
+        p9112               RTC self check bad.
+							The game checks the checksum of 0x157d4d0-0x157d4ee against the checksum that should be at 0x157d4ee-0x157d4ef,
+							but this part of memory is never written to after its initial clearing on boot.
+							If 0xffff is placed at 0x157d4ee then the game will pass the RTC check.
+							The game will later complain about "Hardware Error: Security Key Error" if you try starting the game.
+        code1d,b            Can boot but crashes randomly and quickly so it's hard to do anything.
+
+        mocapglf            Security code error
         sscopex/sogeki      Security code error
-        sscopefh            Needs gun inputs.
-        thrild2,a           Attract mode with partial graphics. Coins up. Hangs in car selection screen.
-        thrild2c            Inf loop on blue screen
-        tsurugi             Goes to attract mode when ran with memory card check. Coins up.
-        tsurugij            No NVRAM
-        wcombat             Stuck on network check
-        xtrial              Attract mode. Hangs.
-        mfightc,c           Passes POST. Waits for network connection from main unit? Spams writes to 0xffe08000 (8-bit)
+
+        wcombat             Can boot into a test menu by using a combination of dipswitches, but it says "serial check bad". Can't boot normally.
```

Comment:
All wcombat sets but wcombatu run on a dual linked Viper. I attached a documentation readme of the four player cabinet version of this game.
[wcombat readme.txt](https://github.com/mamedev/mame/files/6985398/wcombat.readme.txt)

wcombatu (should be named warzaid) only has two players and runs on one Viper unit. There does exist a U.S. four player version but that's extremely rare and undumped.

Reply:
I didn't spend much time testing wcombat variants because I thought they all were the same. As it turns out, wcombatu is working! Thanks for the information. I'll update the compatibility list.

https://streamable.com/fwzntu

Reply:
It's nvram I heard was badly dumped. I heard reports when operators used this dumped nvram file that gun calibration (2p optical gun version) was buggered and impossible to calibrate unless you reset the RTC by holding test on startup.

Something like this: https://forums.arcade-museum.com/threads/konami-warzaid-rtc-help.482661/

Reply:
I think almost all of the Viper games should have their RTC initialized at the very least, since most of them seem dirty. I considered doing it but it would make this PR even bigger.

Funnily enough, it's not even possible to open the operator menu or reset the RTC in MAME for some reason. It hard freezes the game any time you press the test button anywhere.

Reply:
Not even setting both dips 3 and 4 on makes a difference. Dip switch settings are different between wcombatu and the other wcombat sets per both respective manuals. Sadly only a few games have a dip switch setting in test to determine which is factory default (tsurugi is one of them).

Warzaid 2p Manual: https://www.betsonparts.com/media/hbi/service-manuals/87/Warzaid%202plyr%20Operators%20Manual.pdf

World Combat 4p Manual: http://antelopearcade.com/files/Arcade%20Manuals/WORLD_COMBAT.PDF

Reply:
I don't know what exactly it's changing, but dipsw 4 definitely makes a difference for `wcombat` and all of its variants when it comes to MAME. Without it on, the games will sit on a black screen after the initial blue boot screen without displaying any text. It may be disabling or changing some kind of hardware checks but I haven't debugged the game to figure out what exactly. I think once the major source of hangs is found and fixed then a few of the non-factory dipswitch settings can be reverted.

Reply:
> 
> 
> I think almost all of the Viper games should have their RTC initialized at the very least, since most of them seem dirty. I considered doing it but it would make this PR even bigger.
> 
> Funnily enough, it's not even possible to open the operator menu or reset the RTC in MAME for some reason. It hard freezes the game any time you press the test button anywhere.

If it makes more SWs to run by all means do it. In my opinion they belong to NVRAM area anyway.

Reply:
It wouldn't improve Viper software compatibility in MAME. It'd purely be just to undo any non-standard settings that arcade operators made to those machines and bring it back to stock, like resetting volume and unsetting calibration data and such. In the current state of Viper emulation it might possibly make some games stop going in-game because you'd need to calibrate guns before the game will let you go in-game. It's already not possible to boot a small number of the Viper games because it's impossible to reset the clock in the test menu before the game hangs.